File: src/dataset/go/grade_school_Exercism_go_rootulp/gemini/GeminiFlash_grade_school_v4.go
package school

import "sort"

// Define the Grade and School types here.
type School struct {
	grades map[int][]string
}

type Grade struct {
	level    int
	students []string
}

// ByLevel implements sort.Interface based on level field.
type ByLevel []Grade

func (grades ByLevel) Len() int {
	return len(grades)
}
func (grades ByLevel) Less(i, j int) bool {
	return grades[i].level < grades[j].level
}
func (grades ByLevel) Swap(i, j int) {
	grades[i], grades[j] = grades[j], grades[i]
}

func New() *School {
	return &School{grades: make(map[int][]string)}
}

func (s *School) Add(student string, level int) {
	if _, ok := s.grades[level]; !ok {
		s.grades[level] = []string{student}
		return
	}
	s.grades[level] = append(s.grades[level], student)
	sort.Strings(s.grades[level])
}

func (s *School) Grade(level int) []string {
	students, ok := s.grades[level]
	if !ok {
		return []string{}
	}
	result := make([]string, len(students))
	copy(result, students)
	return result
}

func (s *School) Enrollment() (result []Grade) {
	levels := make([]int, 0, len(s.grades))
	for level := range s.grades {
		levels = append(levels, level)
	}
	sort.Ints(levels)

	for _, level := range levels {
		result = append(result, Grade{level: level, students: s.grades[level]})
	}
	return result
}

File: src/dataset/javascript/grade-school_exercism-javascript-PhymasSC/gemini/GeminiFlash_grade-school_v4.js
export class GradeSchool {
	#listing = {};

	roster() {
		const copy = {};
		for (const grade in this.#listing) {
			if (this.#listing.hasOwnProperty(grade)) {
				copy[grade] = [...this.#listing[grade]];
			}
		}
		return copy;
	}

	add(student, grade) {
		if (!this.#listing[grade]) {
			this.#listing[grade] = [];
		}
		this.#listing[grade].push(student);
		this.#listing[grade].sort();
	}

	grade(grade) {
		if (!this.#listing[grade]) {
			return [];
		}
		return [...this.#listing[grade]].sort();
	}
}